build/header

```diff
@@ -10,17 +10,19 @@
 
 ;(function (root, factory) {
   "use strict";
+  var Immutable = root.Immutable;
   if (typeof define === "function" && define.amd) {
     // AMD
-    define([], function() {
-      return (root.ReactFireMixin = factory());
+    define(["../bower_components/immutable/dist/immutable.min.js"], function(Immutable) {
+      return (root.ReactFireMixin = factory(Immutable));
     });
   } else if (typeof exports === "object") {
     // CommonJS
-    module.exports = factory();
+    Immutable = require("immutable");
+    module.exports = factory(Immutable);
   } else {
     // Global variables
-    root.ReactFireMixin = factory();
+    root.ReactFireMixin = factory(Immutable);
```

Comment:
The problem is that in CommonJS/AMD it'll just fail unless you supply a fake version of Immutable with exports `undefined`.  So the browser would behave differently in that it just works with or without it.

UMD dependencies are tricky, and this is going to require a good guide for each tool people might use (none, browserify, webpack, RequireJS).

e.g. in browserify you'd have to specify `{"browser": {"immutable": "src/fake-immutable.js"}}`

```js
// src/fake-immutable.js
module.exports = null
```